src/components/forms/SubmitTemplate/server-action.ts

```diff
@@ -0,0 +1,50 @@
+'use server';
+
+import { getSession, saveTemplate } from '@utils/form-actions';
+import { redirect, RedirectType } from 'next/navigation';
+import { templateFromSessionMapper, validateTemplate } from '@domain/templates';
+import { logger } from '@utils/logger';
+
+export async function submitTemplate(formData: FormData) {
+  const sessionId = String(formData.get('sessionId'));
+
+  const session = await getSession(sessionId);
+
+  if (!session) {
+    return redirect('/invalid-session', RedirectType.push);
+  }
+
+  try {
+    if (session.templateType === 'UNKNOWN') {
+      throw new Error('Unknown template type');
+    }
+
+    const templateDTO = templateFromSessionMapper(
+      session.templateType,
```

Comment:
passing in a field from the session and then passing in the whole session looks odd